stream-access-control/access-control-core/src/main/java/com/backbase/stream/service/UserService.java

```diff
@@ -387,6 +387,18 @@ private Optional<UserRequestBody> updateRequired(EnhancedUserRepresentation curr
             }
         }
 
+        if (!CollectionUtils.isEmpty(user.getAdditionalGroups())) {
+            for (String group : user.getAdditionalGroups()) {
+                if (userRequestBody.getGroups().stream().noneMatch(group::equals)) {
```

Comment:
Isn't it a potential NPE: userRequestBody.getGroups()? I'm not sure, just asking.

Reply:
No, UserRequestBody has following line:
List<String> groups = new ArrayList<>()